From 8058df23560efde4ca9b3e19e77e7bcc577161d9 Mon Sep 17 00:00:00 2001
From: Xulin Sun <xulin.sun@windriver.com>
Date: Sat, 2 Feb 2019 10:50:16 +0800
Subject: [PATCH 2/4] i2c: imx: fix warning 'i2c_imx_init_recovery_info'
 defined but not used

To avoid below warning:
drivers/i2c/busses/i2c-imx.c:1141:12: warning: 'i2c_imx_init_recovery_info'
defined but not used [-Wunused-function]
 static int i2c_imx_init_recovery_info(struct imx_i2c_struct *i2c_imx,
		             ^~~~~~~~~~~~~~~~~~~~~~~~~~

Signed-off-by: Xulin Sun <xulin.sun@windriver.com>
---
 drivers/i2c/busses/i2c-imx.c |    4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/i2c/busses/i2c-imx.c b/drivers/i2c/busses/i2c-imx.c
index 2e402b2..f1d7d76 100644
--- a/drivers/i2c/busses/i2c-imx.c
+++ b/drivers/i2c/busses/i2c-imx.c
@@ -1113,6 +1113,7 @@ static int i2c_imx_xfer(struct i2c_adapter *adapter,
 	return (result < 0) ? result : num;
 }
 
+#ifndef CONFIG_ARCH_LAYERSCAPE
 static void i2c_imx_prepare_recovery(struct i2c_adapter *adap)
 {
 	struct imx_i2c_struct *i2c_imx;
@@ -1177,7 +1178,7 @@ static int i2c_imx_init_recovery_info(struct imx_i2c_struct *i2c_imx,
 
 	return 0;
 }
-
+#else
 /*
  * switch SCL and SDA to their GPIO function and do some bitbanging
  * for bus recovery.
@@ -1221,6 +1222,7 @@ static int i2c_imx_init_recovery_for_layerscape(
 	i2c_imx->layerscape_bus_recover = 1;
 	return 0;
 }
+#endif
 
 static u32 i2c_imx_func(struct i2c_adapter *adapter)
 {
-- 
1.7.9.5

